Gems/LyShine/Code/Source/World/UiCanvasOnMeshComponent.cpp

```diff
@@ -188,36 +237,121 @@ void UiCanvasOnMeshComponent::Deactivate()
     UiCanvasManagerNotificationBus::Handler::BusDisconnect();
 }
 
-bool UiCanvasOnMeshComponent::ProcessCollisionInputEventInternal(const AzFramework::InputChannel::Snapshot& inputSnapshot, const AzFramework::RenderGeometry::RayResult& rayResult)
+////////////////////////////////////////////////////////////////////////////////////////////////////
+bool UiCanvasOnMeshComponent::CalculateUVFromRayIntersection(const AzFramework::RenderGeometry::RayRequest& rayRequest, AZ::Vector2& outUv)
 {
-    AZ::EntityId canvasEntityId = GetCanvas();
+    outUv = AZ::Vector2(0.0f);
+
+    // Make sure we can get the model asset
+    AZ::Data::Asset<AZ::RPI::ModelAsset> modelAsset;
+    AZ::Render::MeshComponentRequestBus::EventResult(
+        modelAsset, GetEntityId(), &AZ::Render::MeshComponentRequestBus::Events::GetModelAsset);
+    AZ::RPI::ModelAsset* asset = modelAsset.Get();
+    if (!asset)
+    {
+        return false;
+    }
 
-    if (canvasEntityId.IsValid())
+    // Calculate the nearest point of collision
+    AZ::Transform meshWorldTM;
+    AZ::TransformBus::EventResult(meshWorldTM, GetEntityId(), &AZ::TransformInterface::GetWorldTM);
+    AZ::Transform meshWorldTMInverse = meshWorldTM.GetInverse();
+
+    AZ::Vector3 rayOrigin = meshWorldTMInverse.TransformPoint(rayRequest.m_startWorldPosition);
+    AZ::Vector3 rayEnd = meshWorldTMInverse.TransformPoint(rayRequest.m_endWorldPosition);
+    AZ::Vector3 rayDirection = rayEnd - rayOrigin;
+
+    float minResultDistance = 2.0f;
```

Comment:
This could use a comment - I'm assuming 2 because the future comparisons will be normalized, but that's purely an educated guess.